src/cmds/core/macro.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from typing import Sequence
+
+import arrow
+from discord import ApplicationContext, Embed, Interaction, SlashCommandGroup, WebhookMessage
+from discord.abc import GuildChannel
+from discord.ext import commands
+from discord.ext.commands import has_any_role
+from sqlalchemy import select
+
+from src.bot import Bot
+from src.core import settings
+from src.database.models import Macro
+from src.database.session import AsyncSessionLocal
+
+logger = logging.getLogger(__name__)
+
+
+class MacroCog(commands.Cog):
+    """Manage macro's."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    macro = SlashCommandGroup("macro", "Manage macro's.", guild_ids=settings.guild_ids)
+
+    @macro.command(description="Add a macro to the records.")
+    @has_any_role(*settings.role_groups.get("ALL_ADMINS"), *settings.role_groups.get("ALL_HTB_STAFF"))
+    async def add(self, ctx: ApplicationContext, name: str, text: str) -> Interaction | WebhookMessage:
+        """Add a macro to the records."""
+        if len(text) == 0:
+            return await ctx.respond("The macro is empty. Try again...", ephemeral=True)
+
+        if len(name) == 0:
+            return await ctx.respond("The macro name is empty. Try again...", ephemeral=True)
+
+        # Name should be lowercase.
+        name = name.lower()
+
+        # We need to check if the macro name already exists.
+        async with AsyncSessionLocal() as session:
+            stmt = select(Macro).filter(Macro.name == name)
+            result = await session.scalars(stmt)
+            macro: Macro = result.first()
+            if macro:
+                return await ctx.respond(f"Macro with the name '{name}' already exists.", ephemeral=True)
```

Comment:
I've changed this so that the column requires the name to be unique. Throws an IntegrityError now and returns the same message. Testing was changed accordingly.